cmd/syft/cli/options/format_cyclonedx_xml_test.go

```diff
@@ -0,0 +1,15 @@
+package options
+
+import (
+	"testing"
+)
+
+func TestFormatCyclonedxXML_buildConfig(t *testing.T) {
+	// assert when building the config that we respond to all possible fields
+
+	ft := FormatCyclonedxXML{}
+	ftp := setAllToNonZero(t, &ft).(*FormatCyclonedxXML)
+
+	subject := ftp.buildConfig("1.2")
+	assertNoZeroFields(t, subject)
```

Comment:
> I think there are a few classes of bugs this doesn't catch

agreed! However, this kind of test is meant to be defensive against future changes: it is difficult to add a field and forget to map it in this function since adding a struct field will cause the test to fail.

Also agreed on input mangling, however, at this time there isn't any transforms being done. I started writing these tests as very simple "here's the input config please assert you get the expected output config" but it was very repetitive and wasn't providing any value against the aforementioned issue (you can easily add a field and forget to map the config).

I think when/if these functions start to add logic (any transformation) then that is a pretty strong signal to add tests around asserting the correct values (which is not being done now).

Reply:
> Maybe there's a deterministic process that setAllToNonZero could use to pick values, that would allow assertNoZeroFields to also assert that there are no transpositions or other changes?

I could base the correct value off of the field name (except for `bool`) which means the value should be verifiable.

Reply:
> Can we use struct embedding to make it so that the version just gets set on the struct?

Not in the current form, that would mean that the package with business logic (down in the `cyclonedxutils` package) would need to import a CLI-only concern (from the `options` package). This import would be in the wrong direction and can open us up to non-obvious package cycles in the future.

Reply:
Based off an offline conversation, I'm keeping the testing approach but I've added the ability for the function to be able to assert the correct value for string types (we can add other types later on an as needed basis).